src/txn/txn_recover.c

```diff
@@ -556,7 +558,8 @@ __wt_txn_recover(WT_SESSION_IMPL *session)
 	 */
 	WT_ERR(session->iface.checkpoint(&session->iface, "force=1"));
 
-done:	FLD_SET(conn->log_flags, WT_CONN_LOG_RECOVER_DONE);
+done:	FLD_SET(conn->log_flags,
+	    WT_CONN_LOG_RECOVER_DIRTY | WT_CONN_LOG_RECOVER_DONE);
```

Comment:
I'm still missing it. Do we need the new call to `FLD_SET` in `__wt_txn_recover` in case `__wt_log_scan` or `checkpoint` return an error, and this latter FLD_SET for other cases? It might be faster for us to talk on the phone next week.